### PR TITLE
Feature/toggle list

### PR DESCRIPTION
**Pull request by DaQuirm**
Extract common functionality into a component: ToggleList (list of `true`/`false` switches that uses arrays of indexes to turn them on/off).
Used by 
* OccurenceIndicator (toggles are matched string patterns (banned characters))
* MatchRenderer (toggles are matched tags)

ToggleList turns `items` into `toggles` (VM is customizable, see below). `pick` is a cell for index arrays.
See tests for more info.
@R1ZZU I hope I haven't broken anything precious :cat:

### CODE REVIEW

**Comment by DaQuirm** [comment]
Closes #90

File: public/common/components/match-renderer/views/index.coffee [hunk 1]
```diff
@@ -13,24 +12,24 @@ MatchRendererView = (vm) ->
 		nxt.Class 'match-renderer'
 		nxt.Attr 'theme', 'default'
 		nxt.Binding vm.tags, (tags = []) ->
-			nxt.Fragment tags.map (tag) ->
-				{ node } = tag
-				if node.type is PuzzleTag.CLOSING
+			nxt.Fragment tags.map (item) ->
+				{tag} = item
```
Comment (kirilknysh) [inline public/common/components/match-renderer/views/index.coffee]: Why don't you like white-spaces? :)

```diff
- {tag} = item
+ { tag } = item
```

**Comment by kirilknysh** [comment]
Are we going to use `toggle-list` somewhere else?

**Comment by kirilknysh** [comment]
Looks OK and working

File: public/common/components/match-renderer/views/index.coffee (hunk 1 above)
Comment (DaQuirm) [inline public/common/components/match-renderer/views/index.coffee]: To differentiate between object literal shorthand and object pattern and it's traditional CoffeeScript syntax (although I write patterns containing nested objects with spaces, like literals:
```coffeescript
{ value: {index} } = data
```

**Comment by DaQuirm** [comment]
Right now I can't think of any use-cases other than these two. But who knows?..

**Pull request merged**